java/ql/test/library-tests/frameworks/apache-collections/test.ql

```diff
@@ -1,16 +1,19 @@
 import java
-import semmle.code.java.dataflow.DataFlow
-import semmle.code.java.dataflow.ExternalFlow
-import semmle.code.java.dataflow.TaintTracking
 import TestUtilities.InlineFlowTest
 
 class SummaryModelTest extends SummaryModelCsv {
   override predicate row(string row) {
     row =
       [
         //"package;type;overrides;name;signature;ext;inputspec;outputspec;kind",
-        // This is temporarily modelled for the helper function newEnumerationWithElement, until the relevant package is modelled
-        "org.apache.commons.collections4.iterators;IteratorEnumeration;true;IteratorEnumeration;;;Element of Argument[0];Element of Argument[-1];value",
+        "generatedtest;Test;false;newWithElement;;;Argument[0];Element of ReturnValue;value",
+        "generatedtest;Test;false;getElement;;;Element of Argument[0];ReturnValue;value",
+        "generatedtest;Test;false;newWithArrayElement;;;Argument[0];ArrayElement of ReturnValue;value",
+        "generatedtest;Test;false;getArrayElement;;;ArrayElement of Argument[0];ReturnValue;value",
+        "generatedtest;Test;false;newWithMapValue;;;Argument[0];MapValue of ReturnValue;value",
+        "generatedtest;Test;false;getMapValue;;;MapValue of Argument[0];ReturnValue;value",
+        "generatedtest;Test;false;newWithMapKey;;;Argument[0];MapKey of ReturnValue;value",
+        "generatedtest;Test;false;getMapKey;;;MapKey of Argument[0];ReturnValue;value",
         "generatedtest;Test;false;newRBWithMapValue;;;Argument[0];MapValue of ReturnValue;value",
         "generatedtest;Test;false;newRBWithMapKey;;;Argument[0];MapKey of ReturnValue;value"
```

Comment:
Are these models necessary? The methods seem to have actual implementations that should be already covered by the dataflow library.

Reply:
I checked and only two are necessary, the ones constructing ResourceBundle (looks like we don't model that yet). I've pushed a commit removing those not required.